packages/ui/src/components/Stepper/Stepper.tsx

```diff
@@ -14,19 +14,19 @@ const deriveStepState = (
   completed = false,
   error = false,
 ): StepState => {
-  if (stepIndex > currentIndex) {
-    return "disabled";
+  if (currentIndex === stepIndex) {
+    return "active";
   }
 
-  if (stepIndex === currentIndex) {
-    return "active";
+  if (error) {
+    return "error";
   }
 
-  if (completed && !error) {
+  if (completed && stepIndex !== currentIndex) {
```

Comment:
Pedantic comment - just like its not necessary to check for `!error` because it would have triggered a return branch, its not necessary to check stepIndex here.

Reply:
lol right, removed in [remove extraneous conditional in stepper state logic](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5902/commits/68615f8bc93a13f551bba7a144ef7fe0dee87ecf)